### PR TITLE
crimson: teuthology suite for seastore

### PR DESCRIPTION
**Pull request by liu-chunmei**
## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by liu-chunmei** [comment]
@athanatos can we let the following PR to run this teuthology test before it get merged?  I only finish readwrite.yaml test, will do rados_api test next.

**Comment by Matan-B** [comment]
> @athanatos can we let each following PR to run this teuthology test before it get merged? I only finish readwrite.yaml test, will do rados_api test next.

IIUC it is possible to scehdule a run with this suite without merging using `--suite-repo` and `--suite-branch` args.
You are welcomed to ping me if this doesn't work out.

**Comment by athanatos** [comment]
@liu-chunmei Yep, you should be able to schedule a run of this prior to merge.

**Comment by liu-chunmei** [comment]
jenkins test make check, please.

**Comment by athanatos** [comment]
retest this please

**Comment by liu-chunmei** [comment]
> @liu-chunmei Yep, you should be able to schedule a run of this prior to merge.

@athanatos how to schedule a run for teuthology test automatically before any following PR get merged? BTW, I can't access redhat Teuthology test environment.

**Comment by liu-chunmei** [comment]
jenkins test make check

**Comment by liu-chunmei** [comment]
jenkins test make check

**Comment by liu-chunmei** [comment]
@athanatos add seastore directory under crimson-rados, please review.

File: qa/suites/crimson-rados/seastore/basic/deploy/cephadm.yaml.disabled [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# no need to verify os + flavor + sha1
```
Comment (athanatos) [inline qa/suites/crimson-rados/seastore/basic/deploy/cephadm.yaml.disabled:1]: The .disabled suffix here indicates that it's disabled, right?  Probably worth removing?

Comment (liu-chunmei) [reply qa/suites/crimson-rados/seastore/basic/deploy/cephadm.yaml.disabled]: done.

**Comment by liu-chunmei** [comment]
jenkins test ceph API tests

**Comment by athanatos** [comment]
Queued run at https://pulpito.ceph.com/sjust-2022-03-04_08:08:42-crimson-rados:seastore-master-distro-default-smithi/

**Comment by liu-chunmei** [comment]
jenkins test make check

**Comment by athanatos** [comment]
make check issues might be fixed with rebase

**Comment by liu-chunmei** [comment]
jenkins test make check.

**Comment by liu-chunmei** [comment]
jenkins test make check.

**Comment by liu-chunmei** [comment]
jenkins test ceph API tests.

**Comment by liu-chunmei** [comment]
> Queued run at https://pulpito.ceph.com/sjust-2022-03-04_08:08:42-crimson-rados:seastore-master-distro-default-smithi/

it is failed.  need filter out ubuntu?  does it only support centos?

**Comment by liu-chunmei** [comment]
@athanatos removed ubuntu, you can run it again.

**Comment by athanatos** [comment]
Scheduled: https://pulpito.ceph.com/sjust-2022-03-09_07:04:10-crimson-rados:seastore-master-distro-default-smithi/

**Comment by liu-chunmei** [comment]
> Scheduled: https://pulpito.ceph.com/sjust-2022-03-09_07:04:10-crimson-rados:seastore-master-distro-default-smithi/

@athanatos (error system:28, No space left on device).

**Comment by athanatos** [comment]
2022-03-09T21:00:51.970 INFO:teuthology.orchestra.run.smithi149.stderr:INFO  2022-03-09 21:00:51,961 [shard 0] seastore_device - block_check_create_device: path=/var/lib/ceph/osd/ceph-0/block, size=107374182400
2022-03-09T21:00:51.970 INFO:teuthology.orchestra.run.smithi149.stderr:ERROR 2022-03-09 21:00:51,962 [shard 0] seastore_device - block_check_create_device: path=/var/lib/ceph/osd/ceph-0/block creation error -- std::system_error (error system:28, No space left on device)

Can we modify it to create seastore devices of half that size?

**Comment by athanatos** [comment]
Queued a run with the seastore device size set to 50G instead of 100G http://pulpito.ceph.com/sjust-2022-03-10_05:59:10-crimson-rados:seastore-master-distro-default-smithi/

**Comment by liu-chunmei** [comment]
> Queued a run with the seastore device size set to 50G instead of 100G http://pulpito.ceph.com/sjust-2022-03-10_05:59:10-crimson-rados:seastore-master-distro-default-smithi/

@athanatos, I changed the default seastore_device_size  to 50G in crimson.yaml.in, and fix a 4 bytes alignment runtime error.   But it still has MGR_MODULE_ERROR, which appear on my teuthology environment in some versions but disappear recently. 
I don't know the reason. 

2022-03-10T20:54:37.322 INFO:tasks.ceph.mgr.x.smithi074.stderr:2022-03-10T20:54:37.316+0000 7f6b357fa700 -1 log_channel(cluster) log [ERR] : Unhandled exception from module 'devicehealth' while running on mgr.x: unknown error
2022-03-10T20:54:37.322 INFO:tasks.ceph.mgr.x.smithi074.stderr:2022-03-10T20:54:37.316+0000 7f6b357fa700 -1 devicehealth.serve:
2022-03-10T20:54:37.322 INFO:tasks.ceph.mgr.x.smithi074.stderr:2022-03-10T20:54:37.316+0000 7f6b357fa700 -1 Traceback (most recent call last):
2022-03-10T20:54:37.323 INFO:tasks.ceph.mgr.x.smithi074.stderr:  File "/usr/share/ceph/mgr/devicehealth/module.py", line 338, in serve
2022-03-10T20:54:37.323 INFO:tasks.ceph.mgr.x.smithi074.stderr:    if self.db_ready() and self.enable_monitoring:
2022-03-10T20:54:37.323 INFO:tasks.ceph.mgr.x.smithi074.stderr:  File "/usr/share/ceph/mgr/mgr_module.py", line 1191, in db_ready
2022-03-10T20:54:37.323 INFO:tasks.ceph.mgr.x.smithi074.stderr:    return self.db is not None
2022-03-10T20:54:37.323 INFO:tasks.ceph.mgr.x.smithi074.stderr:  File "/usr/share/ceph/mgr/mgr_module.py", line 1203, in db
2022-03-10T20:54:37.324 INFO:tasks.ceph.mgr.x.smithi074.stderr:    self._db = self.open_db()
2022-03-10T20:54:37.324 INFO:tasks.ceph.mgr.x.smithi074.stderr:  File "/usr/share/ceph/mgr/mgr_module.py", line 1183, in open_db
2022-03-10T20:54:37.324 INFO:tasks.ceph.mgr.x.smithi074.stderr:    db = sqlite3.connect(uri, check_same_thread=False, uri=True)
2022-03-10T20:54:37.324 INFO:tasks.ceph.mgr.x.smithi074.stderr:sqlite3.DatabaseError: unknown error

2022-03-10T20:54:41.840 INFO:tasks.ceph.mon.a.smithi074.stderr:2022-03-10T20:54:41.841+0000 7f1d3f7fe700 -1 log_channel(cluster) log [ERR] : Health check failed: Module 'devicehealth' has failed: unknown error (MGR_MODULE_ERROR)

**Comment by liu-chunmei** [comment]
@athanatos please reschedule it again to see if fix 4 bytes alignment.  I didn't meet it in my environment.    BTW, any found for the MGR_MODULE_ERROR?

**Comment by athanatos** [comment]
That's a new one on me, @rzarzynski does that ring a bell?

**Comment by athanatos** [comment]
I just pushed that branch to build, will check back in the morning and schedule a run.

**Review by athanatos — APPROVED** [review]

**Comment by athanatos** [comment]
mgr failure seems unrelated, seemed ok to merge

**Pull request merged**